artifactory/services/utils/deleteutils.go

```diff
@@ -96,14 +96,14 @@ func writeRemainCandidate(cw *content.ContentWriter, mergeResult *content.Conten
 	}
 }
 
-func FilterCandidateToBeDeleted(deleteCandidates *content.ContentReader, resultWriter *content.ContentWriter) ([]*content.ContentReader, error) {
+func FilterCandidateToBeDeleted(deleteCandidates *content.ContentReader, resultWriter *content.ContentWriter, candidateType string) ([]*content.ContentReader, error) {
 	paths := make(map[string]ResultItem)
 	pathsKeys := make([]string, 0, utils.MaxBufferSize)
-	dirsToBeDeleted := []*content.ContentReader{}
+	toBeDeleted := []*content.ContentReader{}
 	for candidate := new(ResultItem); deleteCandidates.NextRecord(candidate) == nil; candidate = new(ResultItem) {
-		// Save all dirs candidate in a diffrent temp file.
-		if candidate.Type == "folder" {
-			if candidate.Name == "." {
+		// Save all requested type candidate in a diffrent temp file.
```

Comment:
Save all requested type candidate in a diffrent temp file. -->
Save all candidates, of the requested type, to a different temp file.